airflow/providers/common/sql/hooks/sql.py

```diff
@@ -264,6 +266,7 @@ def run(
                 results = []
                 for sql_statement in sql:
                     self._run_command(cur, sql_statement, parameters)
+                    self._update_query_ids(cur)
```

Comment:
It's an implementation that was before.
For Snowflake, you could it in this PR but for Trino it was deleted by me in this PR
https://github.com/apache/airflow/commit/df00436569bb6fb79ce8c0b7ca71dddf02b854ef#diff-9a702240b5127432b36cea027e9846f74943e8c20c292f4301c182c6063db880

In both cases queryId was extracted after the `execute` and before the handler was applied to the cursor

Reply:
Seems it's synchronous for Trino
https://github.com/trinodb/trino-python-client/blob/b008848101d5eb915c1a7778cf780c72ea61a183/trino/dbapi.py#L407
https://github.com/trinodb/trino-python-client/blob/b008848101d5eb915c1a7778cf780c72ea61a183/trino/client.py#L757

as well as for Snowflake
https://docs.snowflake.com/en/user-guide/python-connector-example.html#performing-a-synchronous-query

Reply:
OK so this solution just won't work.  The on_kill won't actually kill any running query.

Reply:
Does it mean that the original implementation was flawed and we can simply drop it rather than implementing the generic solution for SQLEQO ? 

That certainly loks like it to me.

Reply:
That's how it looks to me.  But I remember @eladkal indicating that the absence of on_kill was preventing some user from migrating to that generic sql operator from trino (though i don't remember where...).  Maybe we can get that user to chime in to confirm?

Reply:
I think i previously used this for snowflake and it worked:
```
    def on_kill(self):
        if self.cnx:
            self.cnx.execute_string(f"SELECT SYSTEM$CANCEL_ALL_QUERIES({self.cnx.session_id})")
            self.cnx.execute_string("ROLLBACK")
```

Reply:
(in a custom hook / operator)

Reply:
In snowflake you can kill query by query_id assuming you know it or by killing all queries of the session.

My point for on_kill() was that if at the end each provider will need it's own on_kill() then it means that we will not have 1 class to serve them all so every provider will need to inherit from SqlExecuteQueryOperator.

I want to avoid all providers doing:
```
class SnowflakeExecuteQueryOperator(SqlExecuteQueryOperator):
    def on_kill():
        ...
```

Reply:
> I want to avoid all providers doing:
> 
> class SnowflakeExecuteQueryOperator(SqlExecuteQueryOperator):
>     def on_kill():
>         ...

it's a very reasonable goal.  at same time, i do think that we have to start from real, working code for the "particular" cases before it will be clear what makes sense in the abstract, general case.  and it seems that we may not have any sql operators that use dbapi hook which properly implement on_kill.

re AsyncSqlExecuteQueryOperator it depends what you're after.

this suggests deferrable operator.  you can have a synchronous sql operator which under the hood submits and polls (asynchronously).  that might be one way to allow for on_kill to function properly but there might also be other ways such as the snowflake example above.  i do think that putting async in name of operator should be determined by the macro operator behavior not the underlying implementation details -- i.e. does it just submit and exit and return query through xcom (or defer).  it's also possible to have an operator param that controls this behavior.

Reply:
I'm OK with taking it one step at a time :)

Reply:
Maybe we could put `on_kill()` logic to the hook's method and just run `hook.kill_query()` in `ExecuteQueryOperator.on_kill()`? 
@eladkal

Reply:
Yeah lets do what we need to get this PR in... Then lets start a followup PR to perfect it